journal/202306.md

```diff
@@ -0,0 +1,861 @@
+- Updated integration test framework (dcrtest) and [re-enabled treasury test](https://github.com/decred/dcrd/pull/3118). It was [temporarily disabled](https://github.com/decred/dcrd/issues/3093) to get around the circular dependency issue.
+- Miners running v1.8 will [generate new block versions](https://github.com/decred/dcrd/pull/3136) (10), and if the majority of the network upgrades [nodes will reject old block versions](https://github.com/decred/dcrd/pull/3135) (9).
+- Modified vote notification logic so that, after block height 777,240, stake voters will [no longer approve blocks with old block versions](https://github.com/decred/dcrd/pull/3137). This is achieved by making dcrd not sent new block notifications to connected RPC clients, such as voting wallets. This change forces miners to upgrade to the latest block version in a timely fashion so that voting on the new consensus changes can commence.
+- Updated the [`AssumeValid`](https://github.com/decred/dcrd/pull/3122) block and [`MinKnownChainWork`](https://github.com/decred/dcrd/pull/3123) as part of the regular release development cycle.
+
+For reference, as of [May 2022](202205.md#development), `AssumeValid` has replaced the legacy "checkpoints" method to optimize initial blockchain downloads. With `AssumeValid`, very old blockchain forks are rejected, and some validation checks are skipped for blocks which are ancestors of the assumed valid block. This can significantly speed up initial blockchain downloads. The updates to `AssumeValid` and `MinKnownChainWork` are externally verified and hardcoded by the devs, and used by default without any action required by node admins. Note: It is possible to change this behavior using the flags `--allowoldforks` and `--assumevalid` as described in `dcrd --help` or [in the code](https://github.com/decred/dcrd/blob/255ef183f66567f1e623026eb524f9cd26fd2b23/chaincfg/mainnetparams.go#L129), but this is not recommended for typical node admins.
+
+{ _what does `MinKnownChainWork` do? can it be disabled like `AssumeValid` and what are the implications?_ }
```

Comment:
It was described a few years back in https://github.com/decred/dcrd/pull/2000.

No, it can't be disabled and there is no reason to since, unlike the old `checkpoints` and rejection of old forks, it can't be invalidated by a chain reorganization.

Reply:
Thanks. I [added](https://github.com/xaur/decred-news/commit/b147200935c98ab6443ec68e43bef8f687e4854e#diff-1d487d8831a0f2f7b2e5f0ea1543054a4e3b287dd31e36d115af3525a053b485R91) an educational paragraph about that.

I used a phrase "minimum known chain work can never go down" and now I'm wondering.

Out of pure theoretical interest, does the "most work" chain always win in Decred given all competing candidates are PoS-approved? Or, can `MinKnownChainWork` get invalidated (i.e. replaced by a smaller value) if PoS voters abandon the "most work" chain and start only approving blocks on a "less work" chain?